core/src/test/java/org/infinispan/security/CacheAuthorizationTest.java

```diff
@@ -91,6 +95,49 @@ public Void run() {
       });
    }
 
+   public void testAdminCombinations() throws Exception {
+      List<Supplier<Void>> calls = Arrays.asList(() -> {
+         cacheManager.getGlobalComponentRegistry();
+         return null;
+      }, () -> {
+         cacheManager.getCacheManagerConfiguration();
+         return null;
+      });
+
+      for (final AuthorizationPermission perm : AuthorizationPermission.values()) {
+         for (Supplier<Void> fn : calls) {
+            Supplier<Void> lazySubjectCall = () -> {
+               try {
+                  Security.doAs(SUBJECTS.get(perm), (PrivilegedExceptionAction<Void>) () -> {
+                     fn.get();
+                     return null;
+                  });
+               } catch (Exception e) {
+                  throw new IllegalStateException(e);
+               }
+               return null;
+            };
+            // only admin must work
+            if (AuthorizationPermission.ADMIN.equals(perm)) {
+               lazySubjectCall.get();
+            } else {
+               try {
+                  lazySubjectCall.get();
+               } catch (Exception e) {
+                  Throwable cause = e.getCause();
+                  if (cause instanceof PrivilegedActionException) {
+                     cause = cause.getCause();
+                  }
+                  boolean isSecurityException = cause instanceof SecurityException;
+                  if (!isSecurityException) {
+                     throw new Exception("Unexpected non-SecurityException", e);
```

Comment:
As this is a test, you should be using the junit assert/fail methods to fail the test.